BackEnd/src/business/stocks.ts

```diff
@@ -0,0 +1,156 @@
+import TeamModel from '../../../Common/models/team.model';
+import { GetDbTeamRosterByTeamId, GetDbTeamsByTeamId } from '../db/teams';
+import { InvalidRequestError } from '../../../Common/errors';
+import { AuthUser } from './auth';
+import { buyDbStockForUser, createDBNewStockValue, getDbLatestStockValue, withdrawDbFunds } from '../db/stocks';
+import { GetRiotLeagueBySummonerId, GetRiotPlayerByPuuid } from '../external-api/player';
+import { RiotLeagueEntryDto } from '../../../Common/Interface/RiotAPI/RiotApiDto';
+import PlayerGameModel from '../../../Common/models/playergame.model';
+import { GetDbPlayerGamesByGameId } from '../db/games';
+import { ToGameId } from '../../../Common/utils';
+import { getDbPlayerTeamPlayerPuuid } from '../db/playerteam';
+import logger from '../../logger';
+
+export async function updateTeamStocksForGame(matchId: string): Promise<void> {
+  logger.info(`Updating Stock Values For Match: ${matchId}`);
+  const allPlayersGames: PlayerGameModel[] = await GetDbPlayerGamesByGameId(ToGameId(matchId));
+
+  // We need to do this because sometimes players arent registered to a team. (Eg, ESUBs)
+  let loserTeamIds: Map<number, number> = new Map<number, number>();
+  let winningTeamIds: Map<number, number> = new Map<number, number>();
+  let seasonIds: Map<number, number> = new Map<number, number>();
+
+  for (let playerGame of allPlayersGames) {
+    const teamId: number = await getDbPlayerTeamPlayerPuuid(playerGame.playerPuuid, playerGame.seasonId);
+    seasonIds.set(playerGame.seasonId, (seasonIds.get(playerGame.seasonId) || 0) + 1);
```

Comment:
I think this whole thing should be refactored into a helper function. Or tbh even better is we actually just do this at ingestion time. This seems to be for determining which teams played?

Reply:
This is being as close to ingestion time as possible. We would need to refactor our models if we wanted to calculate this for the GameModel and i dont want to add that complexity for this change.